src/odemis/gui/win/acquisition.py

```diff
@@ -612,6 +613,409 @@ def on_acquisition_done(self, future):
+        # self.update_acquisition_time(est_time)
+        # self.acq_future = acqmng.acquire(streams, self._main_data_model.settings_obs)
+        self._acq_future_connector = ProgressiveFutureConnector(self.acq_future,
+                                                                self.gauge_acq,
+                                                                self.lbl_acqestimate)
+        self.acq_future.add_done_callback(self.on_acquisition_done)
+
+        self.btn_cancel.SetLabel("Cancel")
+        self.btn_cancel.Bind(wx.EVT_BUTTON, self.on_cancel)
+
+    def on_cancel(self, evt):
+        """ Handle acquisition cancel button click """
+        logging.info("Cancel button clicked, stopping acquisition")
+        self.acq_future.cancel()
+        self.acquiring = False
+        self.btn_cancel.SetLabel("Close")
+        # all the rest will be handled by on_acquisition_done()
+
+    @call_in_wx_main
+    def on_acquisition_done(self, future):
+        """ Callback called when the acquisition is finished (either successfully or cancelled) """
+        if self._main_data_model.opm:
+            self._main_data_model.opm.setAcqQuality(path.ACQ_QUALITY_FAST)
+
+        # bind button back to direct closure
+        self.btn_cancel.Bind(wx.EVT_BUTTON, self.on_close)
+        self._resume_settings()
+
+        self.acquiring = False
+
+        # re-enable estimation time updates
+        self._view.lastUpdate.subscribe(self.on_streams_changed)
+
+        self.acq_future = None  # To avoid holding the ref in memory
+        self._acq_future_connector = None
+
+        try:
+            # TODO: Add code for getting the data from the acquisition future
+            self.data = future.result(1)  # timeout is just for safety
+            self.conf.fn_count = update_counter(self.conf.fn_count)
+        except CancelledError:
+            # put back to original state:
+            # re-enable the acquire button
+            self.btn_secom_acquire.Enable()
+
+            # hide progress bar (+ put pack estimated time)
+            self.update_acquisition_time()
+            self.gauge_acq.Hide()
+            self.Layout()
+            return
+        except Exception:
+            # We cannot do much: just warn the user and pretend it was cancelled
+            logging.exception("Acquisition failed")
+            self.btn_secom_acquire.Enable()
+            self.lbl_acqestimate.SetLabel("Acquisition failed.")
+            self.lbl_acqestimate.Parent.Layout()
+            # leave the gauge, to give a hint on what went wrong.
+            return
+
+        # save result to file
+        self.lbl_acqestimate.Parent.Layout()
+        self.lbl_acqestimate.SetLabel("Acquisition completed.")
+        # As the action is complete, rename "Cancel" to "Close"
+        self._view_data()
+        self.lbl_acqestimate.Parent.Layout()
+
+        # Make sure the file is not overridden
+        self.btn_secom_acquire.Enable()
```

Comment:
As the window is immediately closed, pretty much all of this is not needed.
Just put the content of _view_data() right here, and be done.